src/test/java/org/example/api/FileServiceTest.java

```diff
@@ -0,0 +1,93 @@
+package org.example.api;
+
+import org.example.model.Directory;
+import org.example.model.MyFile;
+import org.example.util.Data;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+class FileServiceTest {
```

Comment:
As for testing now we have 2 possibillites, we can test Service layer and we can also test presentation layer. In general we test both. I want you test both. Service layer test separately and presentation layer tests separately. Let me point out some test cases:
1. create 2 files with same name in 1 directory.
2. create file in a directory which does not exist.
3. remove file from directory that does not exist.
4. remove file which does not exist.
5. copy file to a wrong location.
6. move file to a wrong location.
These are just couple of edge cases, in general you should be covering everything whole code, all lines and functionality you have with tests. This is called test coverage. You should run tests in intelij with "run with coverage" button. It will show you the analytics which lines of code are tested.
![image](https://github.com/Andre-s-Internship/StorageAPI/assets/136591763/4aef15aa-04df-4121-a37c-88b7957b9d1a)

This read lines mean those lines are not tested with the tests, and percentages mean how much of your code is tested. In general we try to test as much as possible in real world. But in our case 100% coverage is achiavable, we don't have much complex things which can not be tested. I want you to reach 100% ocverage of you code with tests.
![image](https://github.com/Andre-s-Internship/StorageAPI/assets/136591763/e0ae1ab2-94a5-45c8-bd3f-32607bbb4339)

Reply:
Added edge cases, also you'll notice some Data.clearSystem() by the end of every test, it is because our Data is static and once it is created it keeps going through all methods and obviously for every method we don't need the modifications of the system from previous tests, that's why I solved that part with this method, which is not in use for the application, but for tests. Also coverage is now 100%, but I didn't implement tests for service yet. Honestly do we actually need it here, as I think if there would be some problem in service, the api will give the errors from there, testing both for me in this case is like testing same thing twice. Obviously if you want me to implement test for Service too, I'll do it, but I don't understand the reason for this case.